pkg/controller/elasticsearch/driver/downscale.go

```diff
@@ -83,8 +83,8 @@ func calculateDownscales(state downscaleState, expectedStatefulSets sset.Statefu
 			}
 
 			toDelete := actualReplicas - expectedReplicas
-			if label.IsMasterNodeSet(actualSset) {
-				toDelete = 1 // Only one removal allowed for masters
+			if label.IsMasterNodeSet(actualSset) && toDelete > 0 {
```

Comment:
I'm not sure to understand why this test (`toDelete > 0`)  is specific to the master case ? It seems odd to have a negative value for any kinds of node.

Reply:
Should we try to add a unit test to demonstrate this corner case ?

Reply:
It's specific because we override `toDelete` only for master. What can happen is that `actualReplicas - expectedReplicas` will be 0 (`0-0`), but due to that check we will set it to 1 anyway and later `targetReplicas` will become -1. I wanted to push the fix out as soon as possible, working on the test right now.

Reply:
Ok, got it. I'm wondering if at some point it would not make sense to implement some `Min` and `Max` function on `int32` instead of these comparisons.